tests/testmain.rs

```diff
@@ -1,78 +1,22 @@
 extern crate lal;
 
-extern crate fs_extra;
 #[macro_use] extern crate log;
-extern crate loggerv;
 extern crate parameterized_macro;
 extern crate tempdir;
-extern crate walkdir;
-
-use std::{
-    env, fs,
-    fs::File,
-    io::prelude::*,
-    path::{Path, PathBuf},
-    process::Command,
-    sync::Once,
-};
 
-use fs_extra::dir::{copy, CopyOptions};
+use std::{fs, path::Path};
+
 use parameterized_macro::parameterized;
-use tempdir::TempDir;
-use walkdir::WalkDir;
 
 use lal::*;
-use loggerv::init_with_verbosity;
-
-struct TestState {
-    backend: LocalBackend,
-    testdir: PathBuf,
-
-    // Keep the tempdir with TestState.
-    // The directory will be cleaned with the TestState is `Drop`ed.
-    tempdir: TempDir,
-}
-
-static START: Once = Once::new();
-
-fn setup() -> TestState {
-    START.call_once(|| {
-        env::set_var("SSL_CERT_FILE", "/etc/ssl/certs/ca-certificates.crt");
-        env::set_var("LAL_CODE_DIR", env::current_dir().unwrap().as_os_str());
-
-        // print debug output and greater from lal during tests
-        init_with_verbosity(2).expect("Setting up test logging");
-    });
-
-    let mut demo_config = PathBuf::from(env::var("LAL_CODE_DIR").unwrap());
-    demo_config.push("./configs/demo.json");
-
-    let mut testdir = PathBuf::from(env::var("LAL_CODE_DIR").unwrap());
-    testdir.push("tests");
-
-    // Do all lal tests in a tempdir as it messes with the manifest
-    let tempdir = TempDir::new("laltest").unwrap();
-
-    let backend = configure_local_backend(&demo_config, &tempdir.path());
-
-    TestState {
-        backend,
-        tempdir,
-        testdir,
-    }
-}
 
-// Copies the component to a temporary location for this test
-// and sets the working directory to that location
-fn clone_component_dir(component: &str, state: &TestState) -> PathBuf {
-    let copy_options = CopyOptions::new();
-
-    let from = state.testdir.join(component);
-    let to = state.tempdir.path().join(component);
-
-    copy(&from, state.tempdir.path(), &copy_options).expect("copy component to tempdir");
-    return to;
-}
+mod common;
+use common::{
+    build_and_stash_update_self, change_envs, check_propagation, clean_check, clone_component_dir,
+    export_check, fetch_release_build_and_publish, get_environment_and_fetch, has_config_and_manifest,
+    init_force, list_everything, no_publish_non_release_builds, query_check, remove_dependencies,
+    run_scripts, setup, shell_echo, shell_permissions, status_on_experimentals, update_save, verify_checks,
+};
```

Comment:
is globbing not more sensible here?

Reply:
There might be a better way to do this in "2018" edition, but I've been holding off thinking too much about the module layouts (and extern crate) until upgrading.